totalcommander.json

```diff
@@ -0,0 +1,57 @@
+{
+    "version": "9.20",
+    "license": "Shareware",
+    "homepage": "https://www.ghisler.com",
+    "description": "Total Commander is a Shareware file manager for Windows® 95/98/ME/NT/2000/XP/Vista/7/8/8.1/10, and Windows® 3.1.",
+    "architecture": {
+        "64bit": {
+            "url": "http://totalcommander.ch/win/tcmd920x64.exe",
+            "hash": "1887c4cb3498572965ad813c6b8c7888514f515b9acba14a6618111db1a43260",
+            "bin": "TOTALCMD64.EXE",
+            "shortcuts": [
+                [
+                    "TOTALCMD64.EXE",
+                    "Total Commander"
+                ]
+            ]
+        },
+        "32bit": {
+            "url": "http://totalcommander.ch/win/tcmd920x32.exe",
+            "hash": "90b7f759df3a7e35c0511b1402508d265704b120622dcd4effb6aa74c238f2c6",
+            "bin": "TOTALCMD.EXE",
+            "shortcuts": [
+                [
+                    "TOTALCMD.EXE",
+                    "Total Commander"
+                ]
+            ]
+        }
+    },
+    "pre_install": [
```

Comment:
Preinstall could be easier when you rewrite installer in first way and then just extract, move.

https://github.com/Ash258/scoop-Ash258/blob/licences/TotalCommander.json#L32-L37

I am creating blank wincmd.key if its not already persisted, because persisting file, which does not exists in first place is quite unexpected. (Need to find time to investigate more).

Reply:
When you persist single file and the file does not exists it will lead to Empty persist folder. And when file is created (by program or manual) its not persisted.

Reply:
Unfortunately, scoop won't create hardlinks and directory junctions unless the files/folders already reside in persist directory. 

There are some other issues I found with current implementation of persist, especially with programs that do not overwrite hardlinked files, but shuffle files by renaming original file by appending .bak (for example) and writing new file from scratch, which will effectively erase hardlink (examples of such program is nomacs, Everything or KeepassXC)

However, in this instance, we could pre-seed configuration files with empty files, and to activate portable mode by setting `UseIniInProgramDir=7` within program's ini file.

Like this:

```json
    "pre_install": [
        "extract_7zip \"$dir\\$fname\" \"$dir\\_tmp\" $false",
        "extract_7zip \"$dir\\_tmp\\INSTALL.CAB\" \"$dir\" $true",
        "Move-Item \"$dir\\e\\*\" \"$dir\"",
        "Remove-Item -Recurse \"$dir\\d\", \"$dir\\e\", \"$dir\\_tmp\", \"$dir\\$fname\"",
        "\"[Configuration]`nUseIniInProgramDir=7\" | Out-File \"$dir\\wincmd.ini\"",
        "ForEach ($emptyFile in $manifest.persist) {New-Item \"$dir\\$emptyFile\" -ItemType file -ErrorAction SilentlyContinue | Out-Null}"
    ],
```

I just tried it, and it doesn't seem to break any functionality of the program. It will overwrite configuration files with its own, as needed.

Reply:
When i am persisting single file which does not exists after installing I am just creating $null or "{}" (For JSOn files). Havent find some spare time to inspect and report whole persist flow. But this look nice.

Reply:
It seems to work with either CR and NL characters. But I will update it to CR.